sys/kern/link_elf.c

```diff
@@ -1733,7 +1733,8 @@ make_capability(const Elf_Sym *sym, caddr_t val)
 	switch (ELF_ST_TYPE(sym->st_info)) {
 	case STT_FUNC:
 	case STT_GNU_IFUNC:
-		val = cheri_andperm(val, CHERI_PERMS_KERNEL_CODE);
+		val = cheri_andperm(val, CHERI_PERMS_KERNEL_CODE |
+		    CHERI_PERM_STORE);
```

Comment:
Yeah this isn't ok, this is something we explicitly don't want. The providers are going to need another way to mess with the code, e.g. via db_write_bytes.

Reply:
Ok, I will replace this patch.

Reply:
I modified FBT to derive a write-only capability from `kernel_root_cap` instead.